source/tests/SGM/SGMEnergyTrajectoryTestView.cpp

```diff
@@ -0,0 +1,392 @@
+    exitSlitPositionTrajectorySeries->setModel(exitSlitPositionTrajectoryData_);
+
+    exitSlitPositionPlot->addItem(exitSlitPositionTheoreticalSeries);
+    exitSlitPositionPlot->addItem(exitSlitPositionTrajectorySeries);
+
+    exitSlitPositionTheoreticalSeries->setLinePen(theoreticalLinePen);
+    exitSlitPositionTheoreticalSeries->setMarker(MPlotMarkerShape::None);
+    exitSlitPositionTheoreticalSeries->setDescription("Theoretical exit slit position");
+
+    exitSlitPositionTrajectorySeries->setLinePen(trajectoryLinePen);
+    exitSlitPositionTrajectorySeries->setMarker(MPlotMarkerShape::None);
+    exitSlitPositionTrajectorySeries->setDescription("Trajectory exit slit position");
+
+    // Exit Slit Position
+    MPlotWidget* energyVsTimePlotWidget = new MPlotWidget();
+    plotStackWidget_->addWidget(energyVsTimePlotWidget);
+    energyVsTimePlotWidget->enableAntiAliasing(true);
+
+    MPlot* energyVsTimePlot = new MPlot(QRect(0,0,320, 480));
+    energyVsTimePlotWidget->setPlot(energyVsTimePlot);
+
+    energyVsTimePlot->axisBottom()->setAxisName("Delta T");
+    energyVsTimePlot->axisLeft()->setAxisName("Resultant Energy");
+
+    energyVsTimePlot->axisScaleLeft()->setAutoScaleEnabled(true);
+    energyVsTimePlot->axisScaleBottom()->setAutoScaleEnabled(true);
+
+    energyVsTimeData_ = new MPlotVectorSeriesData();
+    MPlotSeriesBasic* energyVsTimeSeries = new MPlotSeriesBasic();
+    energyVsTimeSeries->setModel(energyVsTimeData_);
+
+    energyVsTimePlot->addItem(energyVsTimeSeries);
+
+    energyVsTimeSeries->setLinePen(theoreticalLinePen);
+    energyVsTimeSeries->setMarker(MPlotMarkerShape::None);
+    energyVsTimeSeries->setDescription("Energy produced by constant velocity grating angle");
+}
+
+void SGMEnergyTrajectoryTestView::setTheoreticalPlotData(SGMGratingSupport::GratingTranslation gratingTranslation)
+{
+    if(gratingTranslation != SGMGratingSupport::UnknownGrating) {
+        double startValue;
+        double endValue;
+
+        switch(gratingTranslation) {
+        case SGMGratingSupport::LowGrating:
+            startValue = 200;
+            endValue = 800;
+            break;
+        case SGMGratingSupport::MediumGrating:
+            startValue = 400;
+            endValue = 1400;
+            break;
+        case SGMGratingSupport::HighGrating:
+            startValue = 600;
+            endValue = 2500;
+            break;
+        default:
+            startValue = 0;
+            endValue = 0;
+        }
+
+        double increment = 1;
+
+        int numberOfItems = int((endValue - startValue) / increment);
+        SGMMonochromatorInfo monoInfo(200, gratingTranslation);
+
+        QVector<qreal> xValues = QVector<qreal>(numberOfItems);
+        QVector<qreal> gratingAngleYValues = QVector<qreal>(numberOfItems);
+        QVector<qreal> undulatorPositionYValues = QVector<qreal>(numberOfItems);
+        QVector<qreal> exitSlitPositionYValues = QVector<qreal>(numberOfItems);
+
+        xValues.clear();
```

Comment:
I was expecting it to allocate the memory to make room for the elements, but not actually add them (thus preventing the worst case complexity of having to reallocate space when the underlying array has run out of space, and copy over values). I could rewrite this to explicitly overwrite elements at position i, instead of appending.